frontend/src/components/Playback/PlayerElement.vue

```diff
@@ -1,66 +1,180 @@
       <component
-        :is="mediaElement"
+        :is="playbackManager.currentlyPlayingMediaElement"
+        v-show="playbackManager.currentlyPlayingMediaElement === 'video'"
         ref="mediaElementRef"
         :poster="posterUrl"
         autoplay
         crossorigin="anonymous"
         playsinline
-        :loop="playbackManager.isRepeatingOnce" />
+        :loop="playbackManager.isRepeatingOnce"
+        :class="{ stretched: playerElement.isStretched }"
+        @canplay="onCanPlay">
+        <track
+          v-for="sub in playbackManager.currentItemVttParsedSubtitleTracks"
+          :key="`${playbackManager.currentSourceUrl}-${sub.srcIndex}`"
+          kind="subtitles"
+          :label="sub.label"
+          :srclang="sub.srcLang"
+          :src="sub.src" />
+      </component>
     </Teleport>
   </template>
 </template>
 
 <script setup lang="ts">
-import { computed, watch } from 'vue';
+import { computed, watch, ref, nextTick } from 'vue';
 import { isNil } from 'lodash-es';
 import { useI18n } from 'vue-i18n';
-import { playbackManagerStore } from '@/store';
-import { mediaElementRef } from '@/store/playbackManager';
+import Hls, { ErrorData, ErrorTypes, Events } from 'hls.js';
+import { playbackManagerStore, playerElementStore } from '@/store';
 import { getImageInfo } from '@/utils/images';
 import { useSnackbar } from '@/composables';
+import { setNewMediaElementRef } from '@/store/playbackManager';
+
+/**
+ * Temporary calls to set new media element ref until https://github.com/vuejs/core/pull/7593 is fixed
+ */
+const mediaElementRef = ref<HTMLMediaElement>();
+
+setNewMediaElementRef(mediaElementRef);
 
 const playbackManager = playbackManagerStore();
+const playerElement = playerElementStore();
 const { t } = useI18n();
 
-const mediaElement = computed<'audio' | 'video' | undefined>(() => {
-  if (playbackManager.currentlyPlayingMediaType === 'Audio') {
-    return 'audio';
-  } else if (playbackManager.currentlyPlayingMediaType === 'Video') {
-    return 'video';
-  }
-});
+let hls: Hls | undefined;
 
 /**
  * If the player is a video element and we're in the PiP player or fullscreen video playback, we need to ensure the DOM elements are mounted before the teleport target is ready
  */
-const teleportTarget = computed<'body' | '.video-container'>(() =>
-  mediaElement.value === 'audio' ? 'body' : '.video-container'
-);
+const teleportTarget = computed<
+  'body' | '.fullscreen-video-container' | '.minimized-video-container'
+>(() => {
+  if (playbackManager.currentlyPlayingMediaElement === 'audio') {
+    return 'body';
+  } else if (
+    playbackManager.currentlyPlayingMediaElement === 'video' &&
+    playerElement.isFullscreenMounted
+  ) {
+    return '.fullscreen-video-container';
+  } else if (
+    playbackManager.currentlyPlayingMediaElement === 'video' &&
+    playerElement.isPiPMounted
+  ) {
+    return '.minimized-video-container';
+  } else {
+    return 'body';
+  }
```

Comment:
Either:

```suggestion
  if (
    playbackManager.currentlyPlayingMediaElement === 'video' &&
    playerElement.isFullscreenMounted
  ) {
    return '.fullscreen-video-container';
  } else if (
    playbackManager.currentlyPlayingMediaElement === 'video' &&
    playerElement.isPiPMounted
  ) {
    return '.minimized-video-container';
  } else {
    return 'body';
  }
```

Or (more a fan of this one):

```suggestion
  if (playbackManager.currentlyPlayingMediaElement === 'audio') {
    return 'body';
  } else if (playbackManager.currentlyPlayingMediaElement === 'video') {
    if (playerElement.isFullscreenMounted) {
       return '.fullscreen-video-container';
    } else if (playerElement.isPiPMounted) {
       return '.minimized-video-container';
    }
 }
```

I mention it in another comment, but I don't understand why the mediaelement check is now in playbackManager.